src/deadline/client/cli/_groups/asset_group.py

```diff
@@ -237,23 +239,64 @@ def asset_upload(root_dir: str, manifest_dir: str, update: bool, **args):
 @cli_asset.command(name="diff")
 @click.option("--root-dir", help="The root directory to compare changes to. ")
 @click.option(
-    "--manifest", help="The path to manifest folder of the directory to show changes of. "
+    "--manifest-dir",
+    required=True,
+    help="The path to manifest folder of the directory to show changes of. ",
 )
 @click.option(
-    "--format",
-    help="Pretty prints diff information with easy to read formatting. ",
+    "--raw",
+    help="Outputs the raw JSON info of files and their changed statuses. ",
     is_flag=True,
     show_default=True,
     default=False,
 )
 @_handle_error
-def asset_diff(**args):
+def asset_diff(root_dir: str, manifest_dir: str, raw: bool, **args):
     """
-    Check file differences of a directory since last snapshot.
-
-    TODO: show example of diff output
+    Check file differences of a directory since last snapshot, specified by manifest.
     """
-    click.echo("diff shown")
+    if not os.path.isdir(manifest_dir):
+        raise NonValidInputError(f"Specified manifest directory {manifest_dir} does not exist. ")
+
+    if root_dir is None:
+        asset_root_dir = os.path.dirname(manifest_dir)
+    else:
+        if not os.path.isdir(root_dir):
+            raise NonValidInputError(f"Specified root directory {root_dir} does not exist. ")
+        asset_root_dir = root_dir
+
+    # Placeholder Asset Manager
+    asset_manager = S3AssetManager(
```

Comment:
Why placeholder?

Reply:
Sorry, meant as a just a blank Asset Manager to access one of the asset manager functions; this was also something done in snapshot, Jericho and I talked about doing a refactor pass for this